KK_ButtPlugin/ButtplugController.cs

```diff
@@ -164,12 +193,49 @@ override protected IEnumerator Run(int girlIndex)
                     minVibration = 0.6f;
                 }
 
+                // allow caress behavior when female vibrations are enabled
+                if (IsFemale && hscene != null && flags.mode == HFlag.EMode.aibu)
+                {
+                    // for now only add touch vibration to any item between the legs
+                    if (hscene.hand.useItems.Any(item => item?.kindTouch == HandCtrl.AibuColliderKind.kokan || item?.kindTouch == HandCtrl.AibuColliderKind.anal))
+                    {
+                        // TODO have different intensities for different items
+                        var vibration = Mathf.InverseLerp(0, flags.speedMaxItem, flags.speedItem) * Mathf.Lerp(0.2f, 1.0f, flags.speedCalc);
+                        DoVibrate(vibration);
+                        yield return new WaitForSecondsRealtime(1.0f / (float)ButtPlugin.VibrationUpdateFrequency.Value);
+                        continue;
+                    }
+                }
+
                 if (ButtPlugin.SyncVibrationWithAnimation.Value)
                 {
                     // Simple sin based intensity amplification based on normalized position in looping animation
                     var info = animator.GetCurrentAnimatorStateInfo(0);
                     var depth = (info.normalizedTime - GetPhase()) % 1;
                     strength = Mathf.Sin(Mathf.Lerp(0, Mathf.PI, depth)) + 0.1f;
+
+
+                    // masturbation is on a non-speed controlled animation
+                    // it has a fixed order of the animation loops, so we can apply a base strength
+                    //   relative to the intensity of the animation
+                    if (IsFemale && flags.mode == HFlag.EMode.masturbation)

```

Comment:
I accept this block, but nothing else. Sorry.
There _is_ a way to make touch detection portable, but it's complicated and I don't want to include it in the next release.
I would like to do tackle that beast myself when the time comes, if you're okay with that.

Reply:
Plus, I also want to support it for linear toys because it would make sense.

Reply:
That's fine with me.  I don't have a way to test any of the VR stuff, so I'm a bit lost on what's there and what isn't in terms of API.  Is everything else good with the vibration mode selector as long as I remove the touch stuff?  I'm sure there are some players who wish to sync with the female character instead of just the male, so having vibrations during masturbation while also removing them during service is key for that kind of experience.

Reply:
Cheapest way for me to find out would be to just wait until Buttplug starts supporting Switch joy-cons, as they promised they would 😄  But I trust it should work. Animators and HFlags work the same way in desktop and VR. The only VR specific problems I'd ever encountered were all related to the HSceneProc vs VRHScene duality.